### PR TITLE
implement reconnect client.

### PR DESCRIPTION
**Pull request by developeruche**
Various fixes to hyperclient

 - [x] Switch to using [niklasad1/reconnecting-jsonrpsee-ws-client](https://github.com/niklasad1/reconnecting-jsonrpsee-ws-client)
 - [x] Add integration tests for wasm
 - [x] remove loops in `stream::unfold`

### CODE REVIEW

File: modules/client/pkg/package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@polytope-labs/hyperclient",
+  "name": "hyperclient",
```
Comment (seunlanlege) [inline modules/client/pkg/package.json:2]: why?

File: modules/client/pkg/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
   "version": "0.1.0",
```
Comment (seunlanlege) [inline modules/client/pkg/package.json:3]: bump version

File: modules/client/src/interfaces.rs [hunk 3]
```diff
@@ -20,12 +20,20 @@ pub struct JsChainConfig {
     pub hash_algo: i32,
```
Comment (seunlanlege) [inline modules/client/src/interfaces.rs:20]: please remove

**Review by seunlanlege — APPROVED** [review]

File: modules/client/pkg/package.json (hunk 2 above)
Comment (seunlanlege) [reply modules/client/pkg/package.json]: please bump version

**Comment by Wizdave97** [comment]
There's a loop in the `timeout_stream` function also, think we should remove it.

**Review by Wizdave97 — APPROVED** [review]

**Pull request merged**